bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/network/SiemensHvacRequestListener.java

```
@@ -0,0 +1,234 @@
+                            if (error != null) {
+                                errorMsg = error.get("Txt").getAsString();
+                            }
+
+                            if (resultVal) {
+                                hvacConnector.onComplete(result.getRequest());
+                                callback.execute(result.getRequest().getURI(), result.getResponse().getStatus(),
+                                        resultObj);
+
+                                return;
+                            } else if (("datatype not supported").equals(errorMsg)) {
+                                hvacConnector.onComplete(result.getRequest());
+                                callback.execute(result.getRequest().getURI(), result.getResponse().getStatus(),
+                                        resultObj);
+                                return;
+                            } else {
+                                logger.debug("error: {}", subResultObj);
+                                hvacConnector.onError(result.getRequest(), callback);
+                                return;
+                            }
+                        } else {
+                            logger.debug("error: invalid response from gateway, missing subResultObj:Success entry");
+                            hvacConnector.onError(result.getRequest(), callback);
+                            return;
+                        }
+
+                    } else {
+                        logger.debug("error: invalid response from gateway, missing Result entry");
+                        hvacConnector.onError(result.getRequest(), callback);
+                        return;
+                    }
+                }
+            } else {
+                logger.debug("error: content == null");
+                hvacConnector.onError(result.getRequest(), callback);
+                return;
+            }
+        } catch (Exception ex) {
```

Do you need to catch all exceptions here?